spinalcordtoolbox/registration/register.py

```diff
@@ -132,7 +132,20 @@ def register_step_ants_slice_regularized_registration(src, dest, step, metricSiz
     list_fname = [src, dest]
     if fname_mask:
         list_fname.append(fname_mask)
-        mask_options = ['-x', fname_mask]
+        # Check if this mask is soft (i.e. non-binary, such as a Gaussian mask)
+        mask = image.Image(fname_mask)
+        if not np.array_equal(mask.data, mask.data.astype(bool)):
+            # If it is, multiply the target by the soft mask and do not pass
+            # mask options to register() (this is a workaround for ANTS
+            # not supporting non-binary masks).
+            mask_options = []
+            im = image.Image(dest)
+            im_masked = im.copy()
+            im_masked.data = im.data * mask.data
+            im_masked.save()
```

Comment:
so, IIUC the input image is overwritten-- we need to make sure it does not impact the processing down the line (e.g what is the input image is used at another step, without masking?)
having unit tests added would be safer I think

Reply:
This does indeed overwrite the input target image - would it be saver to just save a `dest_masked` as a temporary copy so that the original input target image is unchanged?

Reply:
> would it be safer to just save a dest_masked as a temporary copy so that the original input target image is unchanged?

hum, it depends how the next steps are organized... i would need 15min to dig into the code (no time right now)-- so use this PR for your project if you are in a rush